datanode/src/storage_api.py

```diff
@@ -89,11 +90,47 @@
 @webapp.route('/', methods=['GET'])
 @webapp.route('/status', methods=['GET'])
 def Status():
-  # just a quick status checker, not really a health check
   log.debug('Status handler instantiated...')
-  return _FormatResult({'status': 'success',
-                        'message': 'OK',
-                        'version': VERSION})
+
+  # Check disk usage (make sure not filled with logs)
+  try:
+    usage_stats = os.statvfs('/')
+    usage = round(
+      100 * (1.0 - usage_stats.f_bavail / float(usage_stats.f_blocks)))
+  except EnvironmentError as e:
+    usage = str(e.message)
+
+  # Check ability to perform basic interaction with zookeeper
+  zk_version = wrapper.get_zookeeper_version()
+
+  # Check ability to perform full query from zookeeper
+  try:
+    result = wrapper.get(0, 0, 0)
+    zk_status = 'Unknown issue'
+  except Exception as e:
+    msg = str(e)
+    zk_status = type(e).__name__ + (' ' + msg if msg else '')
+    result = None
+  if result:
+    if result['status'] == 'success':
+      zk_status = 'ok'
+    else:
+      zk_status = result['message']
+
+  return _FormatResult({
+      'status': 'success',
```

Comment:
Should status here be changed to capture the exception, or even change the status if it goes below a certain percentage?

Reply:
Sounds good; done.  HTTP code is still 200 in all cases to indicate that the request for status succeeded, but the JSend changes status to 'error' in off-nominal cases.